packages/aws_common/lib/src/io/aws_file_platform_html.dart

```diff
@@ -145,6 +142,38 @@ class AWSFilePlatform extends AWSFile {
         return blobType;
       });
 
+  @override
+  ChunkedStreamReader<int> getChunkedStreamReader() {
+    return ChunkedStreamReader(stream);
+  }
+
+  @override
+  Stream<List<int>> openRead([int? start, int? end]) {
+    if (super.isStreamOnly) {
```

Comment:
Is this property necessary? I don't like that it's part of the public API and am wondering if we can infer this information without it.

Reply:
You still need to throw InvalidFileException at the end here, so wouldn't it make more sense to `try/catch` things in storage instead of check this flag?

Reply:
Update now infer this info based on `InvalidFileException`.